### PR TITLE
missing language (fr) in default localization.

### PR DESCRIPTION
**Pull request by mjamroz**
I was unable to recreate composer install/update without enabled fr language in laravel-localization (which, by default has only es and en enabled).

### CODE REVIEW

**Comment by mjamroz** [comment]
90995f0 contains search input for searching topos on the crag page.

**Comment by mjamroz** [comment]
d31608 
fixed wrong display of yaxis on sector graphs

File: config/laravellocalization.php [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        //'dua'         => ['name' => 'Duala',                  'script' => 'Latn', 'native' => 'duálá', 'regional' => ''],
+        //'et'          => ['name' => 'Estonian',               'script' => 'Latn', 'native' => 'eesti', 'regional' => 'et_EE'],
+        //'na'          => ['name' => 'Nauru',                  'script' => 'Latn', 'native' => 'Ekakairũ Naoero', 'regional' => ''],
+        //'guz'         => ['name' => 'Ekegusii',               'script' => 'Latn', 'native' => 'Ekegusii', 'regional' => ''],
+        'en'          => ['name' => 'English',                'script' => 'Latn', 'native' => 'English', 'regional' => 'en_GB'],
+        //'en-AU'       => ['name' => 'Australian English',     'script' => 'Latn', 'native' => 'Australian English', 'regional' => 'en_AU'],
+        //'en-GB'       => ['name' => 'British English',        'script' => 'Latn', 'native' => 'British English', 'regional' => 'en_GB'],
+        //'en-US'       => ['name' => 'U.S. English',           'script' => 'Latn', 'native' => 'U.S. English', 'regional' => 'en_US'],
+        'es'          => ['name' => 'Spanish',                'script' => 'Latn', 'native' => 'español', 'regional' => 'es_ES'],
```
Comment (lucien-chastan) [inline config/laravellocalization.php]: comment 'es' language

File: config/laravellocalization.php [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        //'eo'          => ['name' => 'Esperanto',              'script' => 'Latn', 'native' => 'esperanto', 'regional' => ''],
+        //'eu'          => ['name' => 'Basque',                 'script' => 'Latn', 'native' => 'euskara', 'regional' => 'eu_ES'],
+        //'ewo'         => ['name' => 'Ewondo',                 'script' => 'Latn', 'native' => 'ewondo', 'regional' => ''],
+        //'ee'          => ['name' => 'Ewe',                    'script' => 'Latn', 'native' => 'eʋegbe', 'regional' => ''],
+        //'fil'         => ['name' => 'Filipino',               'script' => 'Latn', 'native' => 'Filipino', 'regional' => 'fil_PH'],
+        'fr'          => ['name' => 'French',                 'script' => 'Latn', 'native' => 'français', 'regional' => 'fr_FR'],
```
Comment (lucien-chastan) [inline config/laravellocalization.php]: '**F**rançais' in native key  
(Key value is use in global select language in navbar)

File: config/laravellocalization.php [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    //
+    // IMPORTANT - When hideDefaultLocaleInURL is set to true, the unlocalized root is treated as the applications default locale "app.locale".
+    // Because of this language negotiation using the Accept-Language header will NEVER occur when hideDefaultLocaleInURL is true.
+    //
+    'hideDefaultLocaleInURL' => false,
+
+    // If you want to display the locales in particular order in the language selector you should write the order here. 
+    //CAUTION: Please consider using the appropriate locale code otherwise it will not work
+    //Example: 'localesOrder' => ['es','en'],
+    'localesOrder' => [],
```
Comment (lucien-chastan) [inline config/laravellocalization.php]: `'localesOrder' => ['fr','en'],`

**Comment by mjamroz** [comment]
@lucien-chastan im not sure how pull requests works here in github, but you may find interesting few following commits in this fork: 

f88ac74 

d31608e 

90995f0 (ugly but works)

**Comment by lucien-chastan** [comment]
Hi!
Ideally you should make one branch and one pull request by changes.
in your branch you have 3 changes:
- localization
- the search for the topo
- description corrections
I may or may not merge your branch on the master but I will not choose particular commits.

But it doesn't matter, I agree with your changes.

I'm proposing:

I'll suggest modifications to your branch until it's clean.

Then I'll merge it with the beta branch. we will test on [beta.oblyk](https://beta.oblyk.org/fr) if there is no bug.

Once tested I would merge with the master and push online !!

You're the first real contributor, so we'll find out together how it's gonna go ^^

Thank you so much for your contribution! don't hesitate to come and discuss what you want to change before starting development (for future branches ^^)

**Comment by lucien-chastan** [comment]
For the futur branches, I will be exclud public folder to git, because the pull request is polluate by CSS and JS compiled files

File: .env.example [hunk 4]
```diff
@@ -1,36 +0,0 @@
...
-MAIL_PASSWORD=null
-MAIL_ENCRYPTION=null
-MAIL_FROM_ADDRESS=exemple@mail.com
-MAIL_FROM_NAME=name
-
-PUSHER_APP_ID=
-PUSHER_APP_KEY=
-PUSHER_APP_SECRET=
-
-OBLYK_ID=
```
Comment (lucien-chastan) [inline .env.example]: Please, re-create this file

File: app/Http/Controllers/Chart/SectorChartController.php [hunk 5]
```diff
@@ -81,6 +81,8 @@ function gradeChart($id){
                         [
                             'ticks' => [
                                 'beginAtZero' => true,
+                                'stepSize' => 1,
+                                'step' => 1,
```
Comment (lucien-chastan) [inline app/Http/Controllers/Chart/SectorChartController.php]: Can you show me a screenshot for this change ?

File: app/Http/Controllers/TopoController.php [hunk 6]
```diff
@@ -8,6 +8,8 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\DB;
+use TomLingham\Searchy\Facades\Searchy;
+use Log;
```
Comment (lucien-chastan) [inline app/Http/Controllers/TopoController.php]: I don't think it's used.

File: app/Http/Controllers/Vue/SectorVueController.php [hunk 7]
```diff
@@ -18,9 +18,11 @@ function vueRoutes($id){
...
             ->with(['tickLists' => function ($query) use ($authId) {$query->where('user_id', $authId);}])
             ->with(['crosses' => function ($query) use ($authId) {$query->where('user_id', $authId);}])
-            ->withCount('descriptions')
-            ->withCount('photos')
-            ->withCount('videos')
+            ->withCount([
+                'photos',
+                'videos',
+                'descriptions' => function ($query) { $query->whereNotNull('description');}
+            ])
```
Comment (lucien-chastan) [inline app/Http/Controllers/Vue/SectorVueController.php]: yes, it's more clean

File: resources/assets/js/crag.js [hunk 8]
```diff
@@ -212,7 +212,8 @@ function getSectorChart() {
     for(let i = 0 ; i < sectors.length ; i++){
         let sectorId = sectors[i].getAttribute('data-sector-id');
         axios.get('/chart/sector/' + sectorId + '/grade').then(function (response) {
-            let chart = new Chart(document.getElementById("gradeSectorGraph-" + sectorId).getContext('2d'),JSON.parse(response.data));
+            console.log(response.data);
```
Comment (lucien-chastan) [inline resources/assets/js/crag.js]: for debug yes, remove this after ^^

File: resources/views/modal/topoCrag.blade.php [hunk 9]
```diff
@@ -52,6 +52,11 @@
             </div>
         </div>
 
+            <div class="row">
+                <div class="col s12">
+                    <input onkeyup="getTopoByName()" type="text" placeholder="@lang('modals/paperGuideBook.searchByName')" id="name-search-topo" />
+                </div>
+            </div>
```
Comment (lucien-chastan) [inline resources/views/modal/topoCrag.blade.php]: indentation ?

File: resources/views/pages/crag/partials/liste-topos-search.blade.php [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+                <td><a target="_blank" href="{{route('topoPage',['topo_id'=>$topo->id,'topo_label'=>str_slug($topo->label)])}}">voir le topo</a></td>
+            </tr>
+            @endforeach
+        </tbody>
+    </table>
+@else
+    <p class="text-center grey-text">
+        il n'y a pas de topo 
+    </p>
+@endif
```
Comment (lucien-chastan) [inline resources/views/pages/crag/partials/liste-topos-search.blade.php]: please, add translation

File: app/Http/Controllers/Chart/SectorChartController.php (hunk 5 above)
Comment (mjamroz) [reply app/Http/Controllers/Chart/SectorChartController.php]: https://github.com/lucien-chastan/oblyk/pull/27 here you have cleaner pull request with links to screenshot of this change.

File: .env.example (hunk 4 above)
Comment (mjamroz) [reply .env.example]: i moved stuff to particular pull requests (thanks for the idea!) then there is no need to recreate that one. I will close **this** pull request as soon as i reply to all your comments.

File: app/Http/Controllers/TopoController.php (hunk 6 above)
Comment (mjamroz) [reply app/Http/Controllers/TopoController.php]: Searchy is used (search topo by name) but yes - log is killed already in the new pull request (https://github.com/lucien-chastan/oblyk/pull/29)

File: resources/assets/js/crag.js (hunk 8 above)
Comment (mjamroz) [reply resources/assets/js/crag.js]: killed :gun:

File: resources/views/modal/topoCrag.blade.php (hunk 9 above)
Comment (mjamroz) [reply resources/views/modal/topoCrag.blade.php]: fixed in 9b2e151

File: resources/views/pages/crag/partials/liste-topos-search.blade.php (hunk 10 above)
Comment (mjamroz) [reply resources/views/pages/crag/partials/liste-topos-search.blade.php]: fixed in 9b2e151

**Comment by mjamroz** [comment]
OK, thanks to your suggestions, I made separate pull request for particular functionalities/patches. Im closing this one.

BTW there is some todo list i could work on or you rather dont need my dirty style of coding? 

:mount_fuji:

**Pull request closed without merging**